content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -275,33 +275,79 @@ node {
 }
 ----
 
-=== Defining steps
 
+=== Defining global variables
+
+Internally, scripts in the `vars` directory are instantiated as a singleton
+on-demand, when used first.  So it is possible to define more methods,
+properties on a single file that interact with each other:
+
+[source,groovy]
+----
+// vars/acme.groovy
+def setFoo(v) {
+    this.foo = v;
+}
+def getFoo() {
+    return this.foo;
+}
+def say(name) {
+    echo "Hello world, ${name}"
+}
+----
```

Comment:
Then you can provide a new example in a pull request that you're more happy with :smiling_imp: